src/blindrsa/util.ts

```diff
@@ -0,0 +1,212 @@
+import sjcl from './sjcl';
+
+function assertNever(name: string, x: unknown): never {
+    throw new Error(`unexpected ${name} identifier: ${x}`);
+}
+
+interface HashParams {
+    name: string;
+    hLen: number;
+}
+
+function getHashParams(hash: string): HashParams {
+    switch (hash) {
+        case 'SHA-1':
+            return { name: hash, hLen: 20 };
+        case 'SHA-256':
+            return { name: hash, hLen: 32 };
+        case 'SHA-384':
+            return { name: hash, hLen: 48 };
+        case 'SHA-512':
+            return { name: hash, hLen: 64 };
+        default:
+            assertNever('Hash', hash);
+    }
+}
+
+export function os2ip(bytes: Uint8Array): sjcl.bn {
+    return sjcl.bn.fromBits(sjcl.codec.bytes.toBits(Array.from(bytes)));
+}
+
+export function i2osp(num: sjcl.bn, byteLength: number): Uint8Array {
+    const bytes = new Uint8Array(byteLength);
+    const unpadded = new Uint8Array(sjcl.codec.arrayBuffer.fromBits(num.toBits(undefined), false));
+    bytes.set(unpadded, byteLength - unpadded.length);
+    return bytes;
+}
+
+export function joinAll(a: Uint8Array[]): Uint8Array {
+    let size = 0;
+    for (let i = 0; i < a.length; i++) {
+        size += a[i as number].length;
+    }
+    const ret = new Uint8Array(new ArrayBuffer(size));
+    for (let i = 0, offset = 0; i < a.length; i++) {
+        ret.set(a[i as number], offset);
+        offset += a[i as number].length;
+    }
+    return ret;
+}
+
+export function xor(a: Uint8Array, b: Uint8Array): Uint8Array {
+    if (a.length !== b.length || a.length === 0) {
+        throw new Error('arrays of different length');
+    }
+    const n = a.length;
+    const c = new Uint8Array(n);
+    for (let i = 0; i < n; i++) {
+        c[i as number] = a[i as number] ^ b[i as number];
+    }
+    return c;
+}
+
+function incCounter(c: Uint8Array) {
+    c[3]++;
+    if (c[3] != 0) {
+        return;
+    }
+    c[2]++;
+    if (c[2] != 0) {
+        return;
+    }
+    c[1]++;
+    if (c[1] != 0) {
+        return;
+    }
+    c[0]++;
+}
+
+type MGFFn = (h: HashParams, seed: Uint8Array, mLen: number) => Promise<Uint8Array>;
+
+// MGF1
+// https://www.rfc-editor.org/rfc/rfc8017#appendix-B.2.1
+async function mgf1(h: HashParams, seed: Uint8Array, mLen: number): Promise<Uint8Array> {
```

Comment:
The RFC says to return a `mask too long` error if `maskLen > 2^32 * hLen`: https://www.rfc-editor.org/rfc/rfc8017#appendix-B.2.1. Is it safe to omit that check here?

Reply:
added now